src/storage/addresses/addressesActions.ts

```diff
@@ -93,42 +93,42 @@ export const syncAddressTransactionsNextPage = createAsyncThunk(
   }
 )
 
-export const syncAllAddressesTransactionsNextPage = createAsyncThunk(
-  'addresses/syncAllAddressesTransactionsNextPage',
-  async (_, { getState, dispatch }) => {
+export const syncAddressesTransactionsNextPage = createAsyncThunk(
+  'addresses/syncAddressesTransactionsNextPage',
+  async (
+    { addressHashes, nextPage }: { addressHashes: AddressHash[]; nextPage: number },
+    { getState, dispatch }
+  ): Promise<{ nextPage: number; transactions: Transaction[]; addressHashes: AddressHash[] }> => {
     dispatch(loadingStarted())
 
     const state = getState() as RootState
-    const addresses = selectAllAddresses(state)
-    let nextPage = state.addresses.transactionsPageLoaded
+    const addresses = selectAddresses(state, addressHashes)
+
+    let nextPageToLoad = nextPage
     let newTransactionsFound = false
-    let allTransactionsLoaded = state.addresses.allTransactionsLoaded
```

Comment:
I removed `allTransactionsLoaded` from the addresses state because it doesn't bring any value anymore. The pagination state is handled completely on the Transfers page now.

Reply:
Nice! This make me think: let's rename the "Transactions" widget in the Dashboard page into "Latest transactions"?